core/src/services/icloud/core.rs

```diff
@@ -249,6 +285,18 @@ impl IcloudSigner {
             self.data.scnt = Some(scnt.to_string());
         }
 
+        for (key, value) in resp.headers() {
```

Comment:
Please don't loop the entire headers to find `SET_COOKIE`. Use `get_all` instead.